ark/schema/structure/optional.ts

```diff
@@ -95,14 +96,63 @@ export const Optional = {
 	Node: OptionalNode
 }
 
+const isPrimitive = (value: unknown): value is Primitive =>
+	typeof value === "object" ? value === null : typeof value !== "function"
+const isSimpleSerializeable = (value: unknown): (() => unknown) | false => {
+	if (value instanceof Date) return () => new Date(value)
+	if (
+		Array.isArray(value) &&
+		Object.getPrototypeOf(value) === Array.prototype &&
+		value.every(isPrimitive)
+	)
+		return () => value.slice()
+	if (
+		typeof value === "object" &&
+		value !== null &&
+		Object.getPrototypeOf(value) === Object.prototype &&
+		Object.getOwnPropertySymbols(value).length === 0 &&
+		Object.getOwnPropertyNames(value).every(k => {
+			const prop = Object.getOwnPropertyDescriptor(value, k)
+			return (
+				prop &&
+				"value" in prop &&
+				isPrimitive(prop.value) &&
+				prop.writable === true &&
+				prop.enumerable === true &&
+				prop.configurable === true
+			)
+		})
+	)
+		return () => ({ ...value })
+	return false
+}
+
 export const assertDefaultValueAssignability = (
 	node: BaseRoot,
 	value: unknown,
-	key = ""
+	key: string | null,
+	canOverrideValue: boolean
 ): unknown => {
-	const out = node.in(value)
-	if (out instanceof ArkErrors)
-		throwParseError(writeUnassignableDefaultValueMessage(out.message, key))
+	if (!isPrimitive(value) && typeof value !== "function") {
+		if (!canOverrideValue) {
+			throwParseError(
+				writeNonPrimitiveNonFunctionDefaultValueMessage(key ?? "", value)
+			)
+		}
+		const fn = isSimpleSerializeable(value)
```

Comment:
I don't understand what's going on here. Do we need this? Can't we just accept whatever they return from the function?